File: management-domain/Sales-department/Backend/Nodes/sales-automation-service/commission-calculator-service/src/shared/exceptions/business.exception.ts
import { HttpStatus } from '@nestjs/common';
import { BaseException } from './base.exception';

/**
 * Business logic exception
 * Thrown when a business rule is violated
 */
export class BusinessException extends BaseException {
  readonly errorCode = 'BUSINESS_ERROR';

  constructor(message: string, context?: Record<string, unknown>) {
    super(message, HttpStatus.UNPROCESSABLE_ENTITY, context);
  }
}

File: shared-business-infrastructure/warehousing/warehousing/shared-libraries/Backend/Java/core-libraries/shared-exceptions/src/main/java/com/gogidix/shared/exceptions/domain/exception/BusinessException.java
package com.gogidix.shared.exceptions.domain.exception;

/**
 * Business logic exceptions (4xx errors)
 * Thrown when business rules are violated
 */
public class BusinessException extends BaseException {
    
    public BusinessException(String message) {
        super(message, "BUSINESS_ERROR", 400, "BUSINESS");
    }
    
    public BusinessException(String message, String errorCode) {
        super(message, errorCode, 400, "BUSINESS");
    }
    
    public BusinessException(String message, Throwable cause) {
        super(message, cause, "BUSINESS_ERROR", 400, "BUSINESS");
    }
    
    public BusinessException(String message, String errorCode, int httpStatus) {
        super(message, errorCode, httpStatus, "BUSINESS");
    }

    public BusinessException(String message, Throwable cause, String errorCode, int httpStatus) {
        super(message, cause, errorCode, httpStatus, "BUSINESS");
    }
}
